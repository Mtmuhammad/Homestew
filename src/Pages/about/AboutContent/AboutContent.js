import React from "react";
import "./AboutContent.scss";
import AboutImg from "../AboutImg/AboutImg";
import About1 from "../../../Assets/Images/about/About1.jpg";
import About2 from "../../../Assets/Images/about/About2.jpg";
import About3 from "../../../Assets/Images/about/About3.jpg";
import About4 from "../../../Assets/Images/about/About4.jpg";
import About5 from "../../../Assets/Images/about/About5.jpg";
import About6 from "../../../Assets/Images/about/About6.jpg";
import About7 from "../../../Assets/Images/about/About7.jpg";
import About8 from "../../../Assets/Images/about/About8.jpg";
import About9 from "../../../Assets/Images/about/About9.jpg";
import Culinary from "../../../Assets/Images/about/Culinary.jpg";
import { Link } from "react-router-dom";

const AboutContent = () => {
  return (
    <>
      <div className="col-12 col-lg-6">
        <div className="about-text text-center text-lg-start">
          <h2>What Are We About</h2>
          <p>
            Homestew was created to help users connect through the power and joy
            of food while experiencing the COVID-19 pandemic. We strive to be
            all families' best friends in food and are committed to teaching,
            inspiring, and empowering the every day foodie!
          </p>
          <div className="mt-5 about-btn">
            <Link to="/search">Explore Now</Link>
          </div>
        </div>
      </div>
      <div className="col-12 col-lg-6 mb-5">
        <div className="about-images text-center">
          <div className="row">
            <AboutImg img={About1} />
            <AboutImg img={About2} />
            <AboutImg img={About3} />
            <AboutImg img={About4} />
            <AboutImg img={About5} classes={"d-none d-sm-block"} />
            <AboutImg img={About6} classes={"d-none d-sm-block"} />
            <AboutImg img={About7} classes={"d-none d-sm-block"} />
            <AboutImg img={About8} classes={"d-none d-sm-block"} />
            <AboutImg img={About9} classes={"d-none d-sm-block"} />
          </div>
        </div>
      </div>
      <div
        style={{ marginTop: "8rem", marginBottom: "6rem" }}
        className="d-none d-lg-flex row"
      >
        <div className="d-none d-lg-block col-12 col-lg-6">
          <div className="text-center">
            <img className="img-fluid" src={Culinary} alt="Sushi" />
          </div>
        </div>
        <div className="d-none d-lg-block ps-5 about-list col-12 col-lg-6 text-center text-lg-start">
          <h2>Improve Your Culinary Skills</h2>
          <ul>
            <li>Learn new recipes</li>
            <li>Experiment with foods</li>
            <li>Live a healthier lifestyle</li>
            <li>Understand nutrition facts</li>
            <li>Get cooking tips</li>
            <li>Eat wholesome meals</li>
          </ul>
        </div>
      </div>
    </>
  );
};

export default AboutContent;
