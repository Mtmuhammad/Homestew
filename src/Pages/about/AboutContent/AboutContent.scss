@use "../../../Styles/abstracts/mixins" as *;
@use "../../../Styles/abstracts/variables" as *;

.about-text {
  margin-top: 15rem;

  @include respond(port) {
    margin-top: 6rem;
  }

  & h2 {
    font-size: 4.7rem;
    font-weight: 700;
    letter-spacing: 0.03em;
    padding-right: 20rem;

    @include respond(port) {
      padding-right: 0;
    }
  }

  & p {
    font-size: 1.8rem;
    margin-top: 1.4rem;
    padding-right: 10rem;
    color: $icon-hover-grey;
    line-height: 1.6;

    @include respond(port) {
      padding-right: 0;
    }
  }

  & a {
    color: #fff;
    background-color: $primary-green;
    padding: 1.3rem;
    font-size: 1.5rem;
    text-transform: uppercase;
    border-radius: 5px;
    margin-top: 2rem;
    transition: all 0.3s;

    &:hover {
      color: $primary-font;
    }
  }
}

.about-images {
  margin-top: 8rem;

  @include respond(port) {
    margin-top: 10rem;
  }
}

.about-list {
  margin-top: 3rem;
  & h2 {
    font-size: 4.7rem;
    font-weight: 700;
    letter-spacing: 0.03em;
    padding-right: 5rem;
  }

  & img {
    height: 600px;
  }

  & ul {
    list-style: none;
    padding-left: 0;

    & li {
      font-size: 2rem;
      margin: 1.5rem 0;
      border-left: 4px solid $primary-green;
      padding-left: 0.5rem;
      color: $icon-hover-grey;
    }
  }
}
