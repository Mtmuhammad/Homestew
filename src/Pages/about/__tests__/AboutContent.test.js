import React from "react";
import { render, screen } from "@testing-library/react";
import AboutContent from "../AboutContent/AboutContent";
import {BrowserRouter} from "react-router-dom"
describe("smoke and snapshot tests", () => {
  test("should render without crashing", async () => {
    render(<AboutContent />, {wrapper: BrowserRouter});
  });

  test("should match snapshot", async () => {
    const { asFragment } = render(<AboutContent />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("should render headings", () => {
  test("should render text heading", () => {
    render(<AboutContent />, {wrapper: BrowserRouter});
    const heading = screen.getByRole("heading", { name: /what are we about/i });
    expect(heading).toBeInTheDocument();
  });
  test("should render list heading", () => {
    render(<AboutContent />, {wrapper: BrowserRouter});
    const heading = screen.getByRole("heading", {
      name: /improve your culinary skills/i,
    });
    expect(heading).toBeInTheDocument();
  });
});

test("should render explore button", () => {
  render(<AboutContent />, {wrapper: BrowserRouter});
  const btn = screen.getByRole("link");
  expect(btn).toBeInTheDocument();
  expect(btn).toHaveTextContent(/explore now/i);
});

describe("should render images", () => {
  test("should render text images", () => {
    render(<AboutContent />, {wrapper: BrowserRouter});
    const images = screen.getAllByAltText("About");
    expect(images.length).toBe(9);
    expect(images[0]).toHaveClass("img-fluid");
  });
  test("should render list image", () => {
    render(<AboutContent />, {wrapper: BrowserRouter});
    const image = screen.getByAltText("Sushi");
    expect(image).toBeInTheDocument();
    expect(image).toHaveClass("img-fluid");
  });
});

test("should render list", () => {
  render(<AboutContent />, {wrapper: BrowserRouter});
  const listItems = screen.getAllByRole("listitem");
  expect(listItems.length).toBe(6);
});
