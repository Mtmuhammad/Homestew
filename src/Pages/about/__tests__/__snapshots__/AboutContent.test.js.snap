// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-12 col-lg-6"
  >
    <div
      class="about-text text-center text-lg-start"
    >
      <h2>
        What Are We About
      </h2>
      <p>
        Homestew was created to help users connect through the power and joy of food while experiencing the COVID-19 pandemic. We strive to be all families' best friends in food and are committed to teaching, inspiring, and empowering the every day foodie!
      </p>
      <div
        class="mt-5 about-btn"
      >
        <a
          href="/search"
        >
          Explore Now
        </a>
      </div>
    </div>
  </div>
  <div
    class="col-12 col-lg-6 mb-5"
  >
    <div
      class="about-images text-center"
    >
      <div
        class="row"
      >
        <div
          class="col-6 col-sm-4 "
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About1.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 "
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About2.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 "
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About3.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 "
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About4.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 d-none d-sm-block"
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About5.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 d-none d-sm-block"
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About6.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 d-none d-sm-block"
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About7.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 d-none d-sm-block"
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About8.jpg"
          />
        </div>
        <div
          class="col-6 col-sm-4 d-none d-sm-block"
        >
          <img
            alt="About"
            class="mb-4 about-img img-fluid"
            src="About9.jpg"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="d-none d-lg-flex row"
    style="margin-top: 8rem; margin-bottom: 6rem;"
  >
    <div
      class="d-none d-lg-block col-12 col-lg-6"
    >
      <div
        class="text-center"
      >
        <img
          alt="Sushi"
          class="img-fluid"
          src="Culinary.jpg"
        />
      </div>
    </div>
    <div
      class="d-none d-lg-block ps-5 about-list col-12 col-lg-6 text-center text-lg-start"
    >
      <h2>
        Improve Your Culinary Skills
      </h2>
      <ul>
        <li>
          Learn new recipes
        </li>
        <li>
          Experiment with foods
        </li>
        <li>
          Live a healthier lifestyle
        </li>
        <li>
          Understand nutrition facts
        </li>
        <li>
          Get cooking tips
        </li>
        <li>
          Eat wholesome meals
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
